controllers/base_controller.js

```diff
@@ -186,6 +224,15 @@ module.exports = function BaseControllerModule(pb) {
     };
 
     /**
+     * @method getTemplateService
+     * @param {Object} options for TemplateService
+     * @return {Object} TemplateService
+     */
+    BaseController.prototype.getTemplateService = function(tsOps) {
+        return new pb.TemplateService(tsOps);
+    };
```

Comment:
This function has been removed.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, July 9, 2015 at 5:45 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In controllers/base_controller.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34307463>:

> @@ -186,6 +224,15 @@ module.exports = function BaseControllerModule(pb) {
>      };
>
>      /**
> +     * @method getTemplateService
> +     * @param {Object} options for TemplateService
> +     * @return {Object} TemplateService
> +     */
> +    BaseController.prototype.getTemplateService = function(tsOps) {
> +        return new pb.TemplateService(tsOps);
> +    };


What was the need for this function? Each controller already has an instance of TemplateService created during initialization. Additional instances can be created by calling this.ts.getChildInstance(). The function will return a new instance and provide the same local registrations that the parent instance contained.

Was it just for mocking purposes?

-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34307463>.